serrors/serrors.go

```diff
@@ -17,38 +23,76 @@ func (e *Error) Unwrap() error {
 
 // Error returns the string representation of the error
 func (e *Error) Error() string {
-	return e.error.Error()
+	var elems []string
+	keys := lo.Keys(e.keysAndValues)
+	sort.Strings(keys) // sort keys so we always get a consistent ordering
+	for _, k := range keys {
+		v := e.keysAndValues[k]
+		elems = append(elems, fmt.Sprintf("%s=%v", k, v))
+	}
+	return fmt.Sprintf("%s (%s)", e.error.Error(), strings.Join(elems, ", "))
 }
 
 // WithValues injects additional structured keys and values into the error
 func (e *Error) WithValues(keysAndValues ...any) *Error {
-	e.keysAndValues = append(e.keysAndValues, keysAndValues...)
+	lo.Must0(validateKeysAndValues(keysAndValues))
+	for i := 0; i < len(keysAndValues); i += 2 {
+		e.keysAndValues[keysAndValues[i].(string)] = keysAndValues[i+1]
+	}
 	return e
 }
 
 // Wrap wraps and existing error with additional structured keys and values
 func Wrap(err error, keysAndValues ...any) error {
+	lo.Must0(validateKeysAndValues(keysAndValues))
```

Comment:
nit: do we want to just construct a structured error with on values and call WithValues? Removes a bit of duplication, this is simple enough though so not a big deal either way IMO.